### ISSUE TITLE
`Due Date` not populating on AMD Service Request 

### ISSUE THREAD

**Issue by atdservicebot** [cityofaustin/atd-data-tech#1694]
Jared Wall
AMD

Came by and noticed that a `Due Date` wasn't populating on one of his CSR service requests. 

*Migrated from [atd-knack-data-tracker #240](https://github.com/cityofaustin/atd-knack-data-tracker/issues/240)*

**Comment by atdservicebot** [comment]
*From dianamartin on 2019-11-18*:

Did some initial troubleshooting with Jared. Checked the data publication log to see if script was running. There was an issue on 9/8 and 9/9. The SR was created on 9/8, so that might've been the issue. 

I ran a query on the back end and checked to see if there were any more `Due Dates` not populating and there were a lot that came back blank. 

@johnclary we need to look into this issue. Not sure if it was due to the stop of service for the server migration that Knack did or this was something else.

**Comment by atdservicebot** [comment]
*From johnclary on 2019-11-18*:

@dianamartin thanks for that research. i put this on deck for sprint 12.

**Comment by atdservicebot** [comment]
*From johnclary on 2019-11-18*:

@dianamartin would you check some recent SRs that are missing due dates and see if they have a "due date" flex note?

**Comment by atdservicebot** [comment]
*From dianamartin on 2019-11-18*:

### Spot Testing

`19-00438627` - Closed (Resolved)
created: 11/15/2019
shows due date: 11/3/2020
has flex note `Created Date` field

`19-00441042` - Closed (Resolved)
created: 11/18/2019
does not show due date: `blank`
has flex note `Created Date` field
>Due Date doesn't show up

`19-00440372` - Repairs Complete
created: 11/17/2019
shows due date: 1/6/2020
has flex note `Created Date` field

`19-00423211` - In Progress
created: 11/4/2019
shows due date: 12/19/2019
has flex note `Created Date` field

### User's SR example
`19-00344609` - In Progress
created: 9/8/2019
shows due date: 11/12/2019 
>I manually added this field on the back end with user in the morning during troubleshooting. Referenced due date on the 311 system. `CSR Created Date` didn't match the `Created Date` field though. Not sure if that's an issue or not.

has `NO` flex notes

>We noticed the publication log didn't have a record for 9/8/2019, then on 9/9/2019 showed an error. Will look for SRs during that time period.

### SRs created 9/8 - 9/9
`19-00344616` - Closed (Resolved)
created: 9/8/2019
show due date: 11/12/2019
has flex note `Created Date` field

`19-00344129` - Closed (Duplicate)
created: 9/8/2019
does not show due date: `blank`
has flex note `Created Date` field
>Due Date doesn't show up

`19-00346231` - Closed (Resolved)
created: 9/9/2019
show due date: 10/21/2019
has flex note `Created Date` field

**Comment by atdservicebot** [comment]
*From dianamartin on 2019-11-18*:

@johnclary I've spot checked and even checked a few from the date range that had a publication issue: 9/8/19 and 9/9/2019. It's weird, you can see my notes on what I checked.

**Comment by atdservicebot** [comment]
*From amenity on 2019-12-02*:

Blocked by migration of Knack Proxy server.

**Comment by atdservicebot** [comment]
*From amenity on 2019-12-02*:

@johnclary - can you link this up with the Knack Proxy issues that are blocking it, please?

**Comment by atdservicebot** [comment]
*From dianamartin on 2019-12-10*:

John said that this issue of the `Due Date` field not populating was because the CSR Issue didn't have an associated Flex Note. Apparently, that's where the due date gets generated and plugged into the CSR. 

@johnclary bc of this reason, can we close this issue?

**Comment by atdservicebot** [comment]
*From johnclary on 2019-12-10*:

No, it cannot be closed. We're waiting on Omar to update the ESB to point to new our new Lambda function on AWS. See this issue's dependency.

**Comment by atdservicebot** [comment]
*From dianamartin on 2019-12-10*:

I thought it wasn't related to the knack proxy server migration?

**Comment by atdservicebot** [comment]
*From johnclary on 2019-12-10*:

It is. The on-premise proxy server is occasionally throwing a mysterious error, which we think is related to the server load being too high. The ESB is supposed to keep retrying to send messages when it receives an error, but for some reason the flex notes occasionally do not make it over to Knack.

Our hope is that the new knack proxy server will eliminate these errors because it is more robust.

**Comment by atdservicebot** [comment]
*From dianamartin on 2020-02-05*:

Confirmed with John that this proxy server was migrated. I will need to check the signal CSRs to see if the `Due Date` field is being populated.

**Comment by atdservicebot** [comment]
*From dianamartin on 2020-02-06*:

@johnclary I checked Data Tracker and this is what I noticed: 

`FEB YTD (2/6/2020)`
- 155 records
- 109 blank
- 46 due date field

`JAN`
- 1171 records
- 803 blank
- 368  due date field

I'm not sure if this is a normal amount that have the `Due_Date` field populated or not.

**Comment by atdservicebot** [comment]
*From johnclary on 2020-02-10*:

@dianamartin i made a small change to the integration on Thursday that seems to have resolved the issues with missing flex notes. it won't fix the historical records, but this should be resolved going forward. keep an eye on it and let me know if otherwise. thanks.

**Comment by amenity** [comment]
@johnclary check if there are due dates being added for newer CSRs.

**Comment by dianamartin** [comment]
I just checked the `tmc_issues` that were created this month (506). Of these records 135 of them had the `DUE_DATE` populated, so I think that this is working. 

@johnclary Can we close this issue?

**Comment by johnclary** [comment]
thank you @dianamartin! yes. closed.
